crates/generator/src/syscalls/mod.rs

```diff
@@ -347,26 +350,66 @@ impl<'a, S: State, C: ChainStore, Mac: SupportMachine> Syscalls<Mac> for L2Sysca
                 }
                 let script_hash_prefix = load_bytes(machine, prefix_addr, prefix_len as usize)?;
-                let script_hash = match self
-                    .code_store
-                    .get_script_hash_by_prefix(&script_hash_prefix)
-                {
-                    Some(script_hash) => script_hash,
-                    None => {
-                        machine.set_register(A0, Mac::REG::from_u8(ERROR_NOT_FOUND));
-                        return Ok(true);
+                if let Some(script_hash) = self.get_script_hash_by_prefix(&script_hash_prefix) {
+                    machine
+                        .memory_mut()
+                        .store_bytes(script_hash_addr, script_hash.as_slice())?;
+                    machine.set_register(A0, Mac::REG::from_u8(SUCCESS));
+                } else {
+                    machine.set_register(A0, Mac::REG::from_u8(ERROR_NOT_FOUND));
+                }
+                Ok(true)
+            }
+            SYS_RECOVER_ACCOUNT => {
+                // gw_recover_account(msg: Byte32, signature: Bytes, code_hash: Byte32) -> Script
+                let script_addr = machine.registers()[A0].to_u64();
+                let script_len_addr = machine.registers()[A1].clone();
+                let msg_addr = machine.registers()[A2].to_u64();
+                let signature_addr = machine.registers()[A3].to_u64();
+                let signature_len = machine.registers()[A4].to_u64();
+                let code_hash_addr = machine.registers()[A5].to_u64();
+
+                let msg = load_data_h256(machine, msg_addr)?;
+                let signature = load_bytes(machine, signature_addr, signature_len as usize)?;
+                let code_hash = load_data_h256(machine, code_hash_addr)?;
+
+                if let Some(lock_algo) = self.account_lock_manage.get_lock_algorithm(&code_hash) {
+                    if let Ok(lock_args) = lock_algo.recover(msg, &signature) {
+                        let mut script_args = vec![0u8; 32 + lock_args.len()];
+                        script_args[0..32]
+                            .copy_from_slice(self.rollup_context.rollup_script_hash.as_slice());
+                        script_args[32..32 + lock_args.len()].copy_from_slice(lock_args.as_ref());
+                        let account_script = Script::new_builder()
+                            .code_hash(code_hash.pack())
+                            .hash_type(ScriptHashType::Type.into())
+                            .args(Bytes::from(script_args).pack())
+                            .build();
+
+                        machine.memory_mut().store64(
+                            &script_len_addr,
+                            &Mac::REG::from_u64(account_script.as_slice().len() as u64),
+                        )?;
+                        machine
+                            .memory_mut()
+                            .store_bytes(script_addr, account_script.as_slice())?;
```

Comment:
We should support offset & len args for this syscall.

Reply:
The size is not very large and almost predictable, I think it's meaningless.

Reply:
Ok, We can leave this for now.